plugin/trino-ignite/src/main/java/io/trino/plugin/ignite/IgniteClient.java

```diff
@@ -484,15 +484,11 @@ public Map<String, Object> getTableProperties(Connection connection, JdbcTableHa
         ImmutableMap.Builder<String, Object> properties = ImmutableMap.builder();
         SchemaTableName schemaTableName = tableHandle.asPlainTable().getSchemaTableName();
         String tableName = requireNonNull(schemaTableName.getTableName(), "Ignite table name can not be null").toUpperCase(ENGLISH);
-        String sql = "SELECT idx.CACHE_ID, " +
-                "(SELECT COLUMNS FROM sys.indexes WHERE SCHEMA_NAME = 'PUBLIC' AND TABLE_NAME = ? AND INDEX_NAME = '_key_PK') AS PKS," +
-                "(SELECT COLUMNS FROM sys.indexes WHERE SCHEMA_NAME = 'PUBLIC' AND TABLE_NAME = ? AND INDEX_NAME = 'AFFINITY_KEY') AS AFK FROM sys.indexes as idx " +
-                "JOIN sys.caches che ON idx.CACHE_ID = che.CACHE_ID WHERE idx.SCHEMA_NAME = 'PUBLIC' AND idx.TABLE_NAME = ? LIMIT 1";
+        // Get primary keys from 'sys.indexes' is because DatabaseMetaData.getPrimaryKeys doesn't work well while table being concurrent modified
```

Comment:
```suggestion
        // Get primary keys from 'sys.indexes' because DatabaseMetaData.getPrimaryKeys doesn't work well while table being concurrent modified
```